cpp/test/linalg/axpy.cu

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <raft/linalg/axpy.cuh>
+
+#include "../test_utils.h"
+#include <gtest/gtest.h>
+#include <raft/random/rng.cuh>
+#include <raft/util/cuda_utils.cuh>
+
+namespace raft {
+namespace linalg {
+
+// Reference axpy implementation.
+template <typename T>
+__global__ void naiveAxpy(const int n, const T alpha, const T* x, T* y, int incx, int incy)
+{
+  int idx = threadIdx.x + blockIdx.x * blockDim.x;
+  if (idx < n) { y[idx * incy] += alpha * x[idx * incx]; }
+}
+
+template <typename InType, typename IdxType = int, typename OutType = InType>
+struct AxpyInputs {
+  OutType tolerance;
+  IdxType len;
+  InType alpha;
+  int incx;
+  int incy;
+  unsigned long long int seed;
+};
+
+template <typename T>
+class AxpyTest : public ::testing::TestWithParam<AxpyInputs<T>> {
+ protected:
+  raft::handle_t handle;
+  AxpyInputs<T> params;
+  rmm::device_uvector<T> refy;
+  rmm::device_uvector<T> y;
+
+ public:
+  AxpyTest()
+    : testing::TestWithParam<AxpyInputs<T>>(),
+      refy(0, handle.get_stream()),
+      y(0, handle.get_stream())
+  {
+    handle.sync_stream();
+  }
+
+ protected:
+  void SetUp() override
+  {
+    params = ::testing::TestWithParam<AxpyInputs<T>>::GetParam();
+
+    cudaStream_t stream = handle.get_stream();
+
+    raft::random::RngState r(params.seed);
+
+    int x_len = params.len * params.incx;
+    int y_len = params.len * params.incy;
+    rmm::device_uvector<T> x(x_len, stream);
+    y.resize(y_len, stream);
+
+    uniform(handle, r, x.data(), x_len, T(-1.0), T(1.0));
+    uniform(handle, r, y.data(), y_len, T(-1.0), T(1.0));
+
+    // Take a copy of the random generated values in y for the naive reference implementation
+    // this is necessary since axpy uses y for both input and output
+    refy = rmm::device_uvector<T>(y, stream);
```

Comment:
Instead of re-assigning this instance variable after construction, can we do `refy.resize()` and then use  `raft::copy(refy.data(), y.data())` it makes the intention a little more explicit. This is a common pattern that we use in these tests.

Reply:
sounds good - done in last commit